README.md

```diff
@@ -90,96 +90,104 @@ dist/**
 
 ### jsLint
 
-Type: `Object`.
+Type: `JsLintSettings | Array<{ files?: Array<string>, settings: JsLintSettings }>`.
 
 Configuration for JS linting.
 
-#### jsLint.globals
+`jsLint` can be one of the following:
+
+- A single object that applies to *all* JS files in the codebase.
+- An array of objects that specify different lint settings for different files.
+
+By using an array, you can lint different JS flavors or environments within the same codebase. For example, many codebases have React code for browser within certain directories, Node code in other directories, Jest tests in others, and so on. (This replaces the need for nested and cascading `.eslintrc` files.)
+
+If you use an array, the following rules apply:
+
+- The `files` property of each item is a glob-array that matches files the `settings` apply to. *It is required for every item except the last* (see below).
+- Each file you lint must match *only one* of the `files` glob-arrays. If a file matches multiple, that's a mistake and an error will be thrown.
+- The *last* item in the array does not require a `files` property. If it has no `files` property, it serves as a *fallback*: its `settings` will apply to all files that didn't match one of the items above.
```

Comment:
Nit: It might just be me, but I feel the `last` item convention is not that apparent when simply looking at the config file. If we could somehow indicate somehow that this last object in array is the fallback. 

Feel free to ignore this for now and come back later if we think it is causing trouble with users.

Reply:
Yeah, that's a good critique. I will adjust the documentation so it suggests you use `files: '*'` on your last item to clearly indicate it is a fallback.